pii_recognition/evaluation/metrics.py

```diff
@@ -12,22 +12,24 @@ def compute_f_beta(precision: float, recall: float, beta: float = 1.0) -> float:
 
 
 def compute_label_precision(
-    y_true: List, y_pred: List, label_name: Union[int, str]
+    y_true: List[str], y_pred: List[str], label_name: str,
 ) -> float:
     """Compute recall for a designated label.
 
-    This can calculate precision for a particular label for both binary and
-    multi-class settings.
+    This can calculate precision of a particular label for both binary and multi-class
+    settings. The called upon sklearn function is not stable on string and integer mixed
+    labels, may encouter ValueError. So input arguments are required to be in str.
     """
     return precision_score(y_true, y_pred, average=None, labels=[label_name])[0]
 
 
 def compute_label_recall(
-    y_true: List, y_pred: List, label_name: Union[int, str]
+    y_true: List[str], y_pred: List[str], label_name: str,
```

Comment:
should we cast to strings here? or better to expect that labels are already string?

Reply:
Casting here is another option. I am more towards the later one -- expecting labels are already unicode. Otherwise, we may need to implement comprehensive logics to handle different encodings, i.e., users using non-unicode for predictions and but unicode for labels vice versa. Some other complications once we allowed `y_true` to be a np array. It may better separate the label reconciliation from the actual calculation. What's your opinion on this?

Reply:
is it worth checking that the list type is a string and throwing an exception if not?

Reply:
The type verification is done with `mypy`. In `.travis.yaml` there is a line runs mypy `poetry run mypy --config-file pyproject.toml pii_recognition`, if we pass an unexpected type for an argument, that will raise mypy error.

p.s. made a change to add `TypeVar`, so that `y_true` or `y_pred` can be either a `List[str]` or a `List[int]` but not mix of the two.